modules/client/src/client-common/utils.ts

```diff
@@ -75,3 +80,28 @@ export function isProposalId(propoosalId: string): boolean {
   const regex = new RegExp(/^0x[A-Fa-f0-9]{40}_0x[A-Fa-f0-9]{1,}$/i);
   return regex.test(propoosalId);
 }
+
+export function findLog(
+  receipt: ContractReceipt,
+  iface: Interface,
+  event: string,
+): Log | undefined {
+  return receipt.logs.find(
+    (log) =>
+      log.topics[0] ===
+        id(
+          iface.getEvent(event).format(
+            "sighash",
+          ),
+        ),
+  );
+}
+
+export function getVotingMode(votingSettings: VotingSettings): number {
+  if (votingSettings.earlyExecution) {
```

Comment:
If this function is to be used from subgraph data, I guess it's fine. But if it handles data provided by the user, we should throw an error if both booleans are set to `true`

I would rather encapsulate this logic here, instead of having this check duplicated in other places

Reply:
I was planning on making this check with yup thats why I avoided it here

Reply:
Makes sense. But in my opinion, a function like this should not be allowing an undefined state to return a valid value.

Returning `1` when both are true is an error in my view. Yup may filter one source of errors but we could still use this function from elsewhere and miss a bug